metadata-ingestion/src/datahub/ingestion/glossary/classification_mixin.py

```diff
@@ -130,6 +140,44 @@ def classify_schema_fields(
             self.report.num_tables_classified += 1
             self.populate_terms_in_schema_metadata(schema_metadata, field_terms)
 
+    def update_field_terms(
+        self, field_terms: Dict[str, str], col_info: ColumnInfo
+    ) -> None:
+        term = self.get_terms_for_column(col_info)
+        if term:
+            field_terms[col_info.metadata.name] = term
+
+    def async_classify(
+        self, classifier: Classifier, columns: List[ColumnInfo]
+    ) -> Iterable[ColumnInfo]:
+        num_columns = len(columns)
+        BATCH_SIZE = 5  # Number of columns passed to classify api at a time
+
+        logger.debug(
+            f"Will Classify {num_columns} column(s) with {self.config.classification.max_workers} worker(s) with batch size {BATCH_SIZE}."
+        )
+
+        with concurrent.futures.ProcessPoolExecutor(
+            max_workers=self.config.classification.max_workers,
+        ) as executor:
+            column_info_proposal_futures = [
+                executor.submit(
+                    classifier.classify,
+                    columns[
+                        (i * BATCH_SIZE) : min(i * BATCH_SIZE + BATCH_SIZE, num_columns)
+                    ],
+                )
+                for i in range(ceil(num_columns / BATCH_SIZE))
+            ]
+
+            return [
+                column_with_proposal
+                for proposal_future in concurrent.futures.as_completed(
```

Comment:
as completed can cause the ordering to be inconsistent between runs. is that what we want?

Reply:
I do not anticipate any problems with inconsistent ordering of processing classification columns as each column is independently classified. So this should be okay.